### PR TITLE
fix issue 2968 by adding pragma(framework)

### PR DESCRIPTION
**Pull request by benjones**
For dmd PR: https://github.com/dlang/dmd/pull/10615

### CODE REVIEW

File: spec/pragma.dd [hunk 1]
```diff
@@ -138,6 +139,17 @@ pragma(lib, "foo.lib");
...
 
+$(H3 $(LNAME2 framework, $(D pragma framework)))
+
+     $(P there must be one $(ASSIGNEXPRESSION) and it must evaluate at compile time to a string literal.
+     )
+-----------------
+pragma(framework, "someFramework");
+-----------------
+     $(IMPLEMENTATION_DEFINED
+     Typically, the parameters `-framework someFramework` will be passed to the linker
```
Comment (thewilsonator) [inline spec/pragma.dd]: might want to add "passed to the linker _if applicable_".

**Comment by Geod24** [comment]
The DMD PR has been closed in favor of `linkerDirectives`

**Comment by adamdruppe** [comment]
Just wished I had this again. linkerDirective only works on Windows. framework is a Mac feature.

**Comment by kinke** [comment]
> linkerDirective only works on Windows

That's purely a DMD limitation. `pragma({lib,linkerDirective})` work as embedded Mach-O linker directives with LDC, on Windows *and* Mac.

**Comment by adamdruppe** [comment]
Alas, extern(Objective-C) doesn't work on ldc, so we're at a useless intersection.

**Comment by kinke** [comment]
I don't see how that's related - just saying that there's no object-file limitation to make the existing pragmas work with DMD for Mac too, which cover framework use cases and more.

**Comment by adamdruppe** [comment]
it is related cuz the end goal here requires both, so there's no solution outside patching one or both compilers.

It isn't a difficult patch, but it is hard to realize one's IVY using D when something that was written *14 years ago* never got merged cuz "just use this other thing" and said other thing is also broken.

**Pull request closed without merging**